esp/espcommon/espcommon.cpp

```diff
@@ -0,0 +1,71 @@
+/*##############################################################################
+#    HPCC SYSTEMS software Copyright (C) 2022 HPCC Systems®.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+############################################################################## */
+
+#include "platform.h"
+
+#include "espcommon.hpp"
+
+std::vector<std::string> splitString(const std::string& str, char delim) {
+    std::vector<std::string> strings;
+    size_t start;
+    size_t end = 0;
+    while ((start = str.find_first_not_of(delim, end)) != std::string::npos) {
+        end = str.find(delim, start);
+        strings.push_back(str.substr(start, end - start));
+    }
+    return strings;
+}
+
+// Converts the bucket limit in logical units to the standard units used by the metric framework
+__uint64 convertToBucketLimit(const std::string &units, const std::string &value)
+{
+    __uint64 limit = std::stoull(value);  // the raw value
+    if (units == "s")
+        limit *= 1000000000;
+    else if (units == "ms")
+        limit *= 1000000;
+    else if (units == "us")
+        limit *= 1000;
+
+    return limit;
+}
+
+
+//// function to return the scale factor based on input units
+//ESPCOMMON_API double getUnitsScaleFactor(const std::string &units)
+//{
+//    double unitsScaleFactor = 1;
+//    if (units == "s")
+//        unitsScaleFactor = (double)(1000000000) / getCycleToNanoScale();
+//    else if (units == "ms")
+//        unitsScaleFactor = (double)(1000000) / getCycleToNanoScale();
+//    else if (units == "us")
+//        unitsScaleFactor = (double)(1000) / getCycleToNanoScale();
+//    return unitsScaleFactor;
+//}
+
+// function to return a set of buckets for an execution profile definitions
+ESPCOMMON_API std::vector<__uint64> getExecutionProfileBuckets(const std::string &profileOptions)
+{
+    std::vector<std::string> executionProfileDef = splitString(profileOptions, ',');
```

Comment:
Trivial but FYI we do have existing routines that do this, it would just mean using a StringArray for executionProfileDef instead of a std::vector.  And it would mean not needing to create the splitString function.

```
StringArray executionProfileDef ;
executionProfileDef.appendList(profileOptions.c_str(), ",");
```

Reply:
If it's OK, I'd like to stick with the std lib types.